common/scala/src/main/scala/org/apache/openwhisk/core/connector/Message.scala

```diff
@@ -426,3 +426,16 @@ object EventMessage extends DefaultJsonProtocol {
 
   def parse(msg: String) = Try(format.read(msg.parseJson))
 }
+
+object StatusQuery
+case class StatusData(invocationNamespace: String, fqn: String, waitingActivation: Int, status: String, data: String)
```

Comment:
what is waiting activation?
what is status / why is it a string?
what is data?
are these basic (string & int) types for performance reasons?

Some comments or examples would be helpful here.

Reply:
This is to query the queue status from the scheduler.

The following is an example.
When a `MemoryQueue(AkkaFSM)` is running, there can be a different combination of status and data.
This would be useful to figure out the status when any issue happens in a queue or scheduler.

```
[
...
  {
    "data": "RunningData",
    "fqn": "whisk.system/elasticsearch/status-alarm@0.0.2",
    "invocationNamespace": "style95",
    "status": "Running",
    "waitingActivation": 1
  }
...
]
```

Reply:
The `waitingActivation` is the number of waiting activations in a queue.
Generally, this value would be a small number or zero as most activation messages would be properly handled in time.
But if there is any issue; containers are not provisioned in time, any disconnection with other components happens, etc, there can be many activations waiting for processing.

Regarding the basic types, we just used the string as it is simple.
If we need to change it to certain types with proper serde, how about adding them when we add a feature to use this data class?